pkg/cluster/install.go

```diff
@@ -21,46 +21,82 @@ import (
 
 	"github.com/Azure/ARO-RP/pkg/api"
 	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
 	"github.com/Azure/ARO-RP/pkg/util/restconfig"
 	"github.com/Azure/ARO-RP/pkg/util/steps"
 	"github.com/Azure/ARO-RP/pkg/util/version"
 )
 
 // AdminUpdate performs an admin update of an ARO cluster
 func (m *manager) AdminUpdate(ctx context.Context) error {
-	steps := []steps.Step{
+	isEverything := m.doc.OpenShiftCluster.Properties.MaintenanceTask == api.MaintenanceTaskEverything
+	isOperator := m.doc.OpenShiftCluster.Properties.MaintenanceTask == api.MaintenanceTaskOperator
+
+	// Generic fix-up or setup actions that are fairly safe to always take
+	toRun := []steps.Step{
 		steps.Action(m.initializeKubernetesClients), // must be first
-		steps.Action(m.fixupClusterSPObjectID),
+		steps.Action(m.ensureBillingRecord),         // belt and braces
 		steps.Action(m.ensureDefaults),
-		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.ensureResourceGroup)), // re-create RP RBAC if needed after tenant migration
-		steps.Action(m.createOrUpdateDenyAssignment),
+		steps.NonCriticalAction(m.fixupClusterSPObjectID),
```

Comment:
`NonCriticalAction`  change does not look related to this PR. 
Why are we doing this in the first place?
Like DenyAssignment is for sure critical. This is very confusing now. Very! :D